tests/scripts/generate_config_tests.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""Generate test data for configuration reporting.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+import re
+import sys
+from typing import Iterable, Iterator, List, Optional, Tuple
+
+import scripts_path # pylint: disable=unused-import
+import config
+from mbedtls_dev import test_case
+from mbedtls_dev import test_data_generation
+
+
+def single_option_case(setting: config.Setting, when_on: bool,
```

Comment:
Here we are using "setting" but in the function name thus I would rather go for "single_setting_case" for the function name. That would keep things more aligned with the naming chosen in config.py (config.Setting) I'd say.